dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java

```diff
@@ -574,4 +576,61 @@ public synchronized String suggestVelocityVarName(final String categoryVelVarNam
 	    }
 	}
 
+    /**
+     * Util method to check and generate (if necessary) a unique key for the Category.
+     *
+     * @return {@link Category} with a unique key.
+     */
+    private Category checkUniqueKey(Category category, User user)
+            throws DotDataException, DotSecurityException {
+
+        // If the category is new or if the category doesn't have any key: Let's generate a potential
+        // key and test until we have a unique one.
+        if (!InodeUtils.isSet(category.getInode()) || !UtilMethods.isSet(category.getKey())) {
+            final String potentialKey = getPotentialKeyFromCategory(category);
+            final String uniqueKey = getUniqueKey(potentialKey, user, 1);
+            category.setKey(uniqueKey);
+        } else {
+            // If the category is already in the DB, let's double check that the key is unique,
+            // maybe the the user is editing the category and changing it's key and that key
+            // already used by another Category.
+            final Category categoryInDB = findByKey(category.getKey(), user, false);
+            if (UtilMethods.isSet(categoryInDB)
+                    && !category.getInode().equals(categoryInDB.getInode())) {
+                final String uniqueKey = getUniqueKey(category.getKey(), user, 1);
+                category.setKey(uniqueKey);
+            }
+        }
+        return category;
+    }
+
+	/**
+     * Util method to check is a {@link String} key is unique among the other Category keys. In case
+     * it is repeated ths method will concat "-" + a consecutive number.
+     */
+    private String getUniqueKey(String key, User user, Integer consecutive)
+            throws DotDataException, DotSecurityException {
+
+        if (findByKey(key, user, false) != null) {
+            key = getUniqueKey(key + "-" + consecutive, user, ++consecutive);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "key". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)